### PR TITLE
Replace heavyweight aws-java-sdk with fine-grained plugin

### PR DESCRIPTION
**Pull request by Vlatombe**
Downstream of
* https://github.com/jenkinsci/aws-global-configuration-plugin/pull/20

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
   <properties>
     <revision>1.16</revision>
     <changelist>-SNAPSHOT</changelist>
-    <jenkins.version>2.235.5</jenkins.version>
+    <jenkins.version>2.249.1</jenkins.version>
```
Comment (jglick) [inline pom.xml:19]: Please merge with `master` to get #233.

File: pom.xml [hunk 2]
```diff
@@ -42,7 +42,7 @@
     <dependency>
       <groupId>io.jenkins.plugins</groupId>
       <artifactId>aws-global-configuration</artifactId>
-      <version>1.6</version>
+      <version>1.7-rc87.64ae81df1dd1</version>
```
Comment (jglick) [inline pom.xml:45]:
```suggestion
      <version>1.7</version>
```

File: pom.xml [hunk 3]
```diff
@@ -52,7 +52,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>aws-credentials</artifactId>
-      <version>1.30</version>
+      <version>1.32</version>
```
Comment (jglick) [inline pom.xml:55]: also #233

File: pom.xml [hunk 4]
```diff
@@ -234,6 +234,11 @@
           <groupId>org.slf4j</groupId>
           <artifactId>*</artifactId>
         </exclusion>
+        <exclusion>
+          <!-- use version from jackson2-api -->
+          <groupId>com.fasterxml.jackson.core</groupId>
+          <artifactId>jackson-annotations</artifactId>
+        </exclusion>
```
Comment (jglick) [inline pom.xml:241]: ditto

File: pom.xml [hunk 5]
```diff
@@ -246,8 +251,8 @@
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.235.x</artifactId>
-        <version>918.vae501d2cdc99</version>
+        <artifactId>bom-2.249.x</artifactId>
+        <version>966.v3857b7c82032</version>
```
Comment (jglick) [inline pom.xml:255]: ditto

File: pom.xml [hunk 6]
```diff
@@ -257,9 +262,8 @@
         <version>4.0</version>
       </dependency>
       <dependency>
-        <groupId>org.jenkins-ci.plugins</groupId>
-        <artifactId>aws-java-sdk</artifactId>
-        <version>1.12.70</version>
+        <groupId>org.jenkins-ci.plugins.aws-java-sdk</groupId>
+        <artifactId>aws-java-sdk-minimal</artifactId>
```
Comment (jglick) [inline pom.xml:266]: Probably this is the only change that needs to be here.

File: pom.xml [hunk 7]
```diff
@@ -261,9 +261,8 @@
...
```
Comment (jglick) [inline pom.xml:265]: Maybe I am missing something. Why is this entry, without a version, even here in `dependencyManagement`? It has no effect on `Plugin-Dependencies`, which lists only `aws-credentials` and `aws-global-configuration`, so we are picking up the actual SDK plugin dep only transitively, right?

```
[INFO] io.jenkins.plugins:artifact-manager-s3:hpi:1.16-SNAPSHOT
[INFO] +- io.jenkins.plugins:aws-global-configuration:jar:1.7:compile
[INFO] |  \- org.jenkins-ci.plugins.aws-java-sdk:aws-java-sdk-minimal:jar:1.12.70:compile
[INFO] |     +- com.amazonaws:aws-java-sdk-core:jar:1.12.70:compile
[INFO] |     |  +- software.amazon.ion:ion-java:jar:1.0.2:compile
[INFO] |     |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:jar:2.12.3:compile
[INFO] |     |  \- joda-time:joda-time:jar:2.10.12:compile
[INFO] |     +- com.amazonaws:aws-java-sdk-s3:jar:1.12.70:compile
[INFO] |     |  +- com.amazonaws:aws-java-sdk-kms:jar:1.12.70:compile
[INFO] |     |  \- com.amazonaws:jmespath-java:jar:1.12.70:compile
[INFO] |     +- com.amazonaws:aws-java-sdk-sts:jar:1.12.70:compile
```

And why is S3 a dep of the “minimal” plugin? This is a specific service. I would have expected it to be bundled in some `aws-java-sdk-s3` plugin.

Comment (Vlatombe) [reply pom.xml:265]: See https://github.com/jenkinsci/aws-java-sdk-plugin/pull/540 for context.

Comment (Vlatombe) [reply pom.xml:265]: #239

**Pull request merged**